runtime/common/src/gateway.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright 2023 Centrifuge Foundation (centrifuge.io).
+//
+// This file is part of the Centrifuge chain project.
+// Centrifuge is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version (see http://www.gnu.org/licenses).
+// Centrifuge is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+use cfg_primitives::AccountId;
+use pallet_evm::AddressMapping;
+use polkadot_parachain::primitives::Sibling;
+use sp_core::{crypto::AccountId32, Get, H160};
+use sp_runtime::traits::AccountIdConversion;
+
+use crate::account_conversion::AccountConverter;
+
+pub struct GatewayAccountProvider<T>(core::marker::PhantomData<T>);
+
+impl<T> GatewayAccountProvider<T>
+where
+	T: pallet_evm_chain_id::Config + parachain_info::Config,
+{
+	pub fn get_gateway_account() -> AccountId {
+		let sender_account: AccountId =
```

Comment:
IIUC, that's the parachain's sovereign account?

Reply:
Yes, after talking to @mustermeiszer we figured out that that's the account that we want to derive and then use as the Gateway sender account. We'll have to ensure that both this account and the truncated version of it have enough funds for interacting with the XCM or EVM routers.

Reply:
Yeah it looks like a mimic of that. I don't think we want that, given a recent discussion we had for the `schedule_upgrade` extrinsic of the `LiquidityPools` pallet, where we concluded the treasury account would cover for those fees.

Reply:
> Yes, after talking to @mustermeiszer we figured out that that's the account that we want to derive and then use as the Gateway sender account. We'll have to ensure that both this account and the truncated version of it have enough funds for interacting with the XCM or EVM routers.

@cdamian but then shouldn't we make it a pallet derived account, instead of this mimic of the sovereign account?

Reply:
Ah I see, good point 👍